tahoe_idp/magiclink_helpers.py

```diff
@@ -0,0 +1,45 @@
+from datetime import timedelta
+from uuid import uuid4
+
+from django.conf import settings as djsettings
+from django.http import HttpRequest
+from django.utils import timezone
+from django.utils.crypto import get_random_string
+from tahoe_idp import magiclink_settings as settings
+from tahoe_idp.magiclink_utils import get_client_ip, get_url_path
+from tahoe_idp.models import MagicLink, MagicLinkError
+
+
+def create_magiclink(
+    username: str,
+    request: HttpRequest,
+    redirect_url: str = '',
+) -> MagicLink:
+    limit = timezone.now() - timedelta(seconds=settings.LOGIN_REQUEST_TIME_LIMIT)  # NOQA: E501
```

Comment:
```suggestion
def create_magiclink(
    username: str,
    request: HttpRequest,
    redirect_url: str = '',
    now=None
) -> MagicLink:
    if not now:
        now = timezone.now()
    limit = now - timedelta(seconds=settings.LOGIN_REQUEST_TIME_LIMIT)  # NOQA: E501
```

Reply:
no need for an extra argument, patched in tests